### DISCUSSION TITLE
Update on MacOs with MacPort to 0.40.2, but doesn't have the new features

### DISCUSSION THREAD

**Discussion by Isos9** [jesseduffield/lazygit#3123] [Q&A] [answered]
Hi,

I am trying to update on my mac lazygit, either with homebrew or macports, even though it says 0.40.2, it still shows the old UI, and I don't have the recent features. I tried to restart the shell but nothing.

I can tell because I updated on another device (linux).

Does anyone know why?

Thanks

**Comment by mark2185** [comment]
Does `lazygit --version` say v0.40.2? What does `which lazygit` point to, to the one installed through `brew`?

**Reply by Isos9** [reply]
I tried with brew first, then removed with brew and installed with MacPort.

but in any case, the `--version` always indicated 0.40.2: 
```
 ~ lg ---version
commit=, build date=, build source=macports, version=0.40.2, os=darwin, arch=arm64, git version=2.37.1 (Apple Git-137.1)
```

`which lazygit` return `/opt/local/bin/lazygit`

**Reply by mark2185** [reply]
Okay, and which _new_ UI or features are you lacking? Does the `linux` version also say `v0.42.0`? Could it be that the linux version is built from source and is newer than `v0.40.2`?

**Reply by Isos9** [reply]
I used pacman to install it, I always used a package manager: 

`commit=1d1b8cc01, build date=20231117.063007, build source=unknown, version=v0.40.2.1d1b8cc01, os=linux, arch=amd64, git version=2.37.2`

On Mac I don't see:
- the numbers for each pane at the left
- the new UI that shows the git push or pull aside a branch name instead of the middle of the screen
- the branch delete options (local or remote)
- the branch's upstream options

This is what I used and noticed so far, so I don't know if there is more missing

**Reply by mark2185** [reply]
Well the commit `1d1b8cc01` is from [roughly a month ago](https://github.com/jesseduffield/lazygit/commit/1d1b8cc01f87bb3495426ac8d81d97573f6840d4), whereas tag `v0.40.2` is from 4 months ago.

Did you install it via `pacman` from the official repos, or did you install the `lazygit-git` package via an AUR helper?

**Reply by Isos9** [reply]
Yes you are right, I installed `lazygit-git` via an AUR helper

**Reply by mark2185** [reply]
Well then it was built from source, i.e. `master`, which is _ahead_ of the latest release (`v0.40.2`).

**Reply by Isos9** [reply]
Is it possible to install it similar way with Homebrew or MacPort?

I tried this `brew install jesseduffield/lazygit/lazygit` but didn't change anything

**Reply by stefanhaller** [reply]
Not that I know of.

The easiest way to have a bleeding-edge lazygit is probably to install go (easy with homebrew, e.g. `brew install golang`), and then do `go install github.com/jesseduffield/lazygit@master`).

**Reply by mark2185** [reply]
Sorry, can't really help with that since I don't have a `mac`. The internet does mention the `--build-from-source`, but I'm not sure if that exists.

@stefanhaller IIRC you're on a mac, you might know more.

EDIT: didn't refresh before posting :)

**Reply by stefanhaller** [reply, marked as answer]
Oh wow, I didn't know about the `--build-from-source` argument.

So scratch my earlier answer, this actually works: `brew install --build-from-source --HEAD lazygit`

This deserves to be mentioned in the Readme!

**Reply by Isos9** [reply]
Thanks guys for your help, I used homebrew.
